src/app/components/server/StatusPill.tsx

```diff
@@ -24,71 +24,97 @@ export const StatusPillTypeMap: Record<string, StatusPillType> = {
   Fixed: "fixed",
 };
 
-type DirectTypeProps = { type: StatusPillType };
-type ExposureProps = { exposure: Exposure };
-export type Props = (DirectTypeProps | ExposureProps) & {
+export type Props = {
+  directType?: StatusPillType;
   enabledFeatureFlags?: FeatureFlagName[];
   note?: string;
+  exposure?: Exposure;
+  isRemovalUnderMaintenance?: boolean;
 };
 
 // This component just renders HTML without business logic:
 /* c8 ignore start */
 export const StatusPill = (props: Props) => {
   const l10n = useL10n();
-  const pillType = hasDirectType(props)
-    ? props.type
-    : getExposureStatus(
-        props.exposure,
-        props.enabledFeatureFlags?.includes("AdditionalRemovalStatuses") ??
-          false,
-      );
+
+  // Determine pill type and status label based on props
+  const pillType = props.directType
+    ? props.directType
+    : props.exposure
+      ? getExposureStatus(
+          props.exposure,
+          props.enabledFeatureFlags?.includes("AdditionalRemovalStatuses") ??
+            false,
+          props.isRemovalUnderMaintenance || false,
```

Comment:
Could we break this out to not have a nested ternary condition? I think introducing the extra prop `directType` should not be needed if we already pass `isRemovalUnderMaintenance` to `getExposureStatus`.

Additionally, I’m wondering why we can’t check `removal_under_maintenance` on the exposure as we did before — or should we remove the status check in the switch statement?

Reply:
But couldn’t we (instead of adding additional logic to the status pill), keep it unchanged and pass `type={removal_under_maintenance}` for broker results under removal?